forum/forum/routing.py

```diff
@@ -0,0 +1,24 @@
+from django.urls import path
+from channels.routing import URLRouter
+from channels.generic.websocket import AsyncWebsocketConsumer
+
+
+class Consumer(AsyncWebsocketConsumer):
+    async def connect(self):
+        await self.accept()
+        await self.send(text_data="Websocket connected.")
+
+    async def receive(self, text_data):
+        await self.send(text_data=f"You sent: {text_data}")
+
+    async def disconnect(self):
+        pass
+
+websocket_urlpatterns =[
+    path("ws/test/", Consumer.as_asgi()),
+]
+
+application = URLRouter(websocket_urlpatterns)    
```

Comment:
Invalid application object
The routing.py file **should not export URLRouter directly as application**. It should be **part of** the ProtocolTypeRouter in asgi.py, as you already did there:
application = ProtocolTypeRouter({
    "http": get_asgi_application(),
    "websocket": URLRouter(websocket_urlpatterns)
})